front/src/applications/stdcm/components/StdcmResults/StdcmResults.tsx

```diff
@@ -93,6 +98,56 @@ const StdcmResults = ({
     return extractMarkersInfo(outputs.results.simulationPathSteps);
   }, [hasSimulationResults, outputs]);
 
+  const [similarTrains, setSimilarTrains] = useState<SimilarTrainWithSecondaryCode[]>([]);
+
+  const [postSimilarTrains] = osrdEditoastApi.endpoints.postSimilarTrains.useMutation();
+
+  useEffect(() => {
+    const searchForSimilarTrains = async () => {
+      const { consist, pathSteps } = selectedSimulation.inputs;
+      if (!consist) {
+        return;
+      }
+
+      const key = (uic: number | undefined, ch: string) => `${uic}-${ch}`;
+
+      const isStopByOpKey = pathSteps.reduce((acc, ps) => {
+        const k = key(ps.location?.uic, ps.location?.secondary_code ?? '');
+        acc.set(k, !ps.isVia || ps.stopFor !== undefined);
+        return acc;
+      }, new Map<string, boolean>());
+
+      const waypoints =
+        (selectedSimulation.outputs?.pathProperties.manchetteOperationalPoints ?? []).map((op) => {
+          const k = key(op.extensions?.identifier?.uic, op.extensions?.sncf?.ch ?? '');
+          return {
+            id: op.opId!,
+            stop: isStopByOpKey.get(k) ?? false,
+          };
+        }) ?? [];
```

Comment:
You don't need this anymore
```suggestion
        });
```

Reply:
done